streams/src/main/java/org/apache/kafka/streams/state/internals/NamedCache.java

```diff
@@ -358,28 +359,46 @@ private void update(LRUCacheEntry entry) {
         final StreamsMetricsImpl metrics;
         final String groupName;
         final Map<String, String> metricTags;
+        final Map<String, String> allMetricTags;
         final Sensor hitRatioSensor;
 
+        private void maybeAddMetric(Sensor sensor, MetricName name, MeasurableStat stat) {
+            if (!metrics.metrics().containsKey(name)) {
+                sensor.add(name, stat);
+            } else {
+                log.trace("Trying to add metric twice: {}", name);
+            }
+        }
 
         public NamedCacheMetrics(StreamsMetrics metrics) {
             final String scope = "record-cache";
-            final String entityName = name;
             final String opName = "hitRatio";
             final String tagKey = "record-cache-id";
-            final String tagValue = name;
+            final String tagValue = ThreadCache.underlyingStoreNamefromCacheName(name);
```

Comment:
It is better to keep `tagKey = scope + "-id";` in case we change `scope` in the future.